### PR TITLE
add autocomplete to click decorators

### PR DESCRIPTION
**Pull request by maxandersen**
Fixes: #2678

### CODE REVIEW

**Comment by maxandersen** [comment]
here is an initial fix - I tried look for places to put test but haven't found it yet.

File: third_party/2and3/click/decorators.pyi [hunk 1]
```diff
@@ -83,7 +83,8 @@ def argument(
     metavar: Optional[str] = ...,
     expose_value: bool = ...,
     is_eager: bool = ...,
-    envvar: Optional[Union[str, List[str]]] = ...
+    envvar: Optional[Union[str, List[str]]] = ...,
+    autocompletion: Callable[[Any, List, str], List[Union[str, Tuple]] = ...,
```
Comment (JelleZijlstra) [inline third_party/2and3/click/decorators.pyi]: Can you add type arguments to List and Tuple? What is in the Lists and Tuples?

Comment (maxandersen) [reply third_party/2and3/click/decorators.pyi]: done.

File: third_party/2and3/click/decorators.pyi [hunk 2]
```diff
@@ -83,7 +83,8 @@ def argument(
     metavar: Optional[str] = ...,
     expose_value: bool = ...,
     is_eager: bool = ...,
-    envvar: Optional[Union[str, List[str]]] = ...
+    envvar: Optional[Union[str, List[str]]] = ...,
+    autocompletion: Callable[[Any, List[str]], str], List[Union[str, Tuple[str, str]]] = ...,
```
Comment (srittau) [inline third_party/2and3/click/decorators.pyi]: This should be `Optional` as it defaults to `None` in `Parameter.__init__()`. Also, it seems the brackets are misaligned, which is also why CI fails.

**Comment by JelleZijlstra** [comment]
I tried to count them but I guess I am bad at counting :)

**Pull request merged**